### PR TITLE
Fix CentOS 7 image build on aarch64

### PR DESCRIPTION
**Pull request by mbautin**
CentOS 7 is an important OS for our CI/CD pipeline, probably with the oldest version of glibc of any supported Linux distribution as of 10/2022. By building software on CentOS 7, we can produce packages that work on most glibc-supported Linux distributions. This patch enables building a CentOS 7 Docker image for aarch64.

Software version changes:
- Using Golang version 1.19.2 on aarch64 CentOS 7, by installing from a tarball and not from a repository. We should probably switch to this way of installing Golang across the board, for all other Linux versions as well.
- Updating the version of Python installed from source from 3.8.2 to 3.8.14.

Also generating more locales on RHEL-based systems. Previously, even the en_US.UTF-8 locale was not being generated on CentOS 7. The list of locales was provided by @andrei-mart based on what is needed by Postgres for internationalization.

### CODE REVIEW

File: docker_setup_scripts/centos.sh [hunk 1]
```diff
@@ -129,7 +130,11 @@ install_packages() {
   local toolset_package_suffixes=( "${TOOLSET_PACKAGE_SUFFIXES_COMMON[@]}" )
   if [[ $centos_major_version -eq 7 ]]; then
     toolset_prefix="devtoolset"
-    gcc_toolsets_to_install=( "${CENTOS7_GCC_TOOLSETS_TO_INSTALL[@]}" )
+    if [[ $( uname -m ) == "x86_64" ]]; then
+      gcc_toolsets_to_install=( "${CENTOS7_GCC_TOOLSETS_TO_INSTALL_X86_64[@]}" )
+    else
```
Comment (SrivastavaAnubhav) [inline docker_setup_scripts/centos.sh:135]: Do we want to have `else if` here or is it always going to be aarch?

File: docker_setup_scripts/centos.sh [hunk 2]
```diff
@@ -188,11 +193,38 @@ install_packages() {
...
+    (
+      cd "${tmp_dir}"
+      wget "${go_url}"
+      actual_sha256=$( sha256sum "${go_archive_name}" | awk '{print $1}' )
+      if [[ ${actual_sha256} != "${expected_sha256}" ]]; then
+        echo >&2 "Invalid SHA256 sum of ${go_archive_name}: expected ${expected_sha256}, got" \
+                 "${actual_sha256}"
+        exit 1
+      fi
+      tar xzf "${go_archive_name}"
```
Comment (SrivastavaAnubhav) [inline docker_setup_scripts/centos.sh:212]: Missing a `-`?

File: docker_setup_scripts/centos.sh (hunk 1 above)
Comment (mbautin) [reply docker_setup_scripts/centos.sh:135]: Right now yes, we only support x86_64 and aarch64.

File: docker_setup_scripts/centos.sh (hunk 2 above)
Comment (mbautin) [reply docker_setup_scripts/centos.sh:212]: This works fine, this is how I always invoke tar.

**Pull request merged**